src/main/java/com/webonise/tomcat8/redisession/RedisPoolFactory.java

```diff
@@ -0,0 +1,94 @@
+package com.webonise.tomcat8.redisession;
+
+import java.util.Enumeration;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.NamingException;
+import javax.naming.RefAddr;
+import javax.naming.Reference;
+import javax.naming.spi.ObjectFactory;
+
+import com.webonise.tomcat8.redisession.redisclient.Redis;
+
+import redis.clients.jedis.Protocol;
+
+public class RedisPoolFactory implements ObjectFactory {
+
+	  public static final String DEFAULT_KEY_HOST = "host";
+	  public static final String DEFAULT_KEY_PORT = "port";
+	  public static final String DEFAULT_KEY_TIMEOUT =  "timeout";
+	  private static Logger logger = Logger.getLogger(RedisPoolFactory.class.getName());
+	
+	  
+	  /**
+	   * Extracts properties from Resources
+	   */
+  public Object getObjectInstance(Object obj,
+      Name name, Context nameCtx, Hashtable<?,?> environment)
+      throws NamingException {
+
+      // Acquire an instance of our specified bean class
+      //
+      final Map<String,Object> jedisConfig = new Hashtable<String,Object>();
+      // Customize the bean properties from our attributes
+      Reference ref = (Reference) obj;
+      Enumeration addrs = ref.getAll();
```

Comment:
Move your `Enumeration` to a `List` using [Collections](http://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#list-java.util.Enumeration-). Please also use the appropriate generic type parameters, so you don't have to do a bunch of unsafe casting (your IDE should be complaining about all this).